fastapi/routing.py

```diff
@@ -404,18 +404,22 @@ def __init__(
         self.response_class = response_class
 
         assert callable(endpoint), "An endpoint must be a callable"
-        self.dependant = get_dependant(path=self.path_format, call=self.endpoint)
-        for depends in self.dependencies[::-1]:
-            self.dependant.dependencies.insert(
-                0,
-                get_parameterless_sub_dependant(depends=depends, path=self.path_format),
-            )
+        self.dependant = self.gen_dependant()
         self.body_field = get_body_field(dependant=self.dependant, name=self.unique_id)
         self.dependency_overrides_provider = dependency_overrides_provider
         self.callbacks = callbacks
         self.app = request_response(self.get_route_handler())
         self.openapi_extra = openapi_extra
 
+    def gen_dependant(self):
+        dependant = get_dependant(path=self.path_format, call=self.endpoint)
+        for depends in self.dependencies[::-1]:
+            dependant.dependencies.insert(
+                0,
+                get_parameterless_sub_dependant(depends=depends, path=self.path_format),
```

Comment:
Assign it to a variable and pass the bound variable to the function.

Reply:
It just dry the code from init function and reuse the code block to generate self.dependant

Reply:
Are you still review this PR?

Reply:
No.

Reply:
why ? I can't understand

Reply:
You said it is just dry. What do you want me to do now? Approve the changes?

Reply:
Well, it was you to ask me why I change this code and I answered it.  Now I curious that did you actually read  this PR's title and content. I think i write it correctly, and there is a related issues shown in the description.

And without any further information just leave like you haven't been here. What a greatest reviewer you are.

Reply:
#3524 and this is the related issue, if you cannot find it in the PR's content